Sources/SwiftParser/TokenConsumer.swift

```diff
@@ -23,6 +23,13 @@ protocol TokenConsumer {
   /// Consume the current token and change its token kind to `remappedTokenKind`.
   mutating func consumeAnyToken(remapping remappedTokenKind: RawTokenKind) -> Token
 
```

Comment:
Do we ever take a prefix where it's not just the default text of the token? Could we just pass the token in rather than the prefix as well?

Reply:
Yes, there’s

https://github.com/apple/swift-syntax/blob/e990c2a4b6f1e00ff000ec06fd672d8d224a5753/Sources/SwiftParser/Availability.swift#L288

and

https://github.com/apple/swift-syntax/blob/e990c2a4b6f1e00ff000ec06fd672d8d224a5753/Sources/SwiftParser/Declarations.swift#L1115

Good idea though, I added a `precondition` to check that `prefix` is `tokenKind.defaultText` if that `tokenKind` has a `defaultText`.

Reply:
If they're the minority, could still be worth having a `consumePrefix` that only takes the token kind instead. Maybe this is more clear as to what's happening though 🤷

Reply:
The problem is that this method doesn’t have a clearly defined meaning if `defaultText == nil`. I prefer being explicit here and passing the prefix.